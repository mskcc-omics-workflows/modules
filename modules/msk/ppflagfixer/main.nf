process PPFLAGFIXER {
    tag "$meta.id"
    label 'process_medium'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'docker://mskcc/facets-suite:2.0.9':
        'docker.io/mskcc/facets-suite:2.0.9' }"

    input:
    tuple val(meta), path(process_bam), path(process_bam_index)

    output:
    tuple val(meta), path("ppflag.*.bam")   , emit: ppflag_bam
    path "versions.yml"                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    /usr/bin/ppflag-fixer \
        ${args} \
        ${process_bam} \
        ppflag.${prefix}.bam

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        htslib: \$(bgzip --version | grep -oP '(?<=\\(htslib\\) ).*')
        r: \$(R --version | grep -oP '(?<=R version ).*(?=\\()')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """
    echo "stub test" >> ppflag.${prefix}.bam
    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        htslib: \$(bgzip --version | grep -oP '(?<=\\(htslib\\) ).*')
        r: \$(R --version | grep -oP '(?<=R version ).*(?=\\()')
    END_VERSIONS
    """
}
