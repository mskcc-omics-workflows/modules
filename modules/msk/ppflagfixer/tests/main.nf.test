nextflow_process {

    name "Test PPFLAGFIXER"
    script "../main.nf"
    process "PPFLAGFIXER"
    config "./nextflow.config"

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "ppflagfixer"

    test("ppflagfixer - normal bam") {

        when {
            process {
                """
                    meta = [ id:'test' ] // meta map
                    input[0] = [
                                    meta,
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_normal_bam'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_normal_bam_bai'], checkIfExists: true)
                                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("ppflagfixer - tumor bam") {

        when {
            process {
                """
                    meta = [ id:'test' ] // meta map
                    input[0] = [
                                    meta,
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_tumor_bam'], checkIfExists: true),
                                    file(params.test_data_mskcc['bam_files']['test_argos_sample_1_tumor_bam_bai'], checkIfExists: true)
                                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("ppflagfixer - bam - stub") {

        options "-stub"

        when {
            process {
                """
                    meta = [ id:'test' ] // meta map
                    input[0] = [
                                    meta,
                                    file("stub_file"),
                                    file("stub_file1")
                                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
