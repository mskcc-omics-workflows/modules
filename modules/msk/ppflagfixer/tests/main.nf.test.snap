{
    "ppflagfixer - normal bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "ppflag.test.bam:md5,d25c13205b1d33c5c4b12999fc839d86"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ed3820a8eb6f0db2010ddec9aefa7249"
                ],
                "ppflag_bam": [
                    [
                        {
                            "id": "test"
                        },
                        "ppflag.test.bam:md5,d25c13205b1d33c5c4b12999fc839d86"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ed3820a8eb6f0db2010ddec9aefa7249"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.2"
        },
        "timestamp": "2024-02-26T23:05:23.642445"
    },
    "ppflagfixer - bam - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "ppflag.test.bam:md5,3d77e3e110a219d3495d44e243450313"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ed3820a8eb6f0db2010ddec9aefa7249"
                ],
                "ppflag_bam": [
                    [
                        {
                            "id": "test"
                        },
                        "ppflag.test.bam:md5,3d77e3e110a219d3495d44e243450313"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ed3820a8eb6f0db2010ddec9aefa7249"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.2"
        },
        "timestamp": "2024-02-26T23:06:14.360871"
    },
    "ppflagfixer - tumor bam": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "ppflag.test.bam:md5,d05041f7678abb654e5d349ec430b131"
                    ]
                ],
                "1": [
                    "versions.yml:md5,ed3820a8eb6f0db2010ddec9aefa7249"
                ],
                "ppflag_bam": [
                    [
                        {
                            "id": "test"
                        },
                        "ppflag.test.bam:md5,d05041f7678abb654e5d349ec430b131"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ed3820a8eb6f0db2010ddec9aefa7249"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.04.2"
        },
        "timestamp": "2024-02-26T23:06:08.243018"
    }
}